src/github.com/matrix-org/dendrite/common/config/config.go

```diff
@@ -189,6 +190,17 @@ type Dendrite struct {
 		// The config for the jaeger opentracing reporter.
 		Jaeger jaegerconfig.Configuration `yaml:"jaeger"`
 	}
+
+	// Any information derived from the configuration options for later use.
+	Derived struct {
+		// Flows for registration. As long as they given flows only relies on config file options,
+		// we can generate them on startup and store them until needed
+		Flows []authtypes.Flow `json:"flows"`
+
+		// Params for registration. Data that is returned to the client while registering and
+		// that which is necessary to complete certain registration flow stages
+		Params map[string]interface{} `json:"params"`
+	}
```

Comment:
We may want to stick this in a `Registration` sub structure, otherwise its not clear that `config.Derived.Flows` is a registration thing

Reply:
+1.

TBH I'm not entirely sure this belongs in the Config object at all. Shouldn't it be up to the register.go module to figure out the right flows based on the config?

Reply:
The intention was to generate these structures once on startup, then use them over and over again during each registration.

We could place the actual calculation in register.go, though we would simply be calling it from config.go after the config file was read.